src/configmypy/argparse_config.py

```diff
@@ -62,7 +62,13 @@ def read_conf(self, config=None, **additional_config):
         parser = argparse.ArgumentParser(description='Read the config from the commandline.')
         for key, value in iter_nested_dict_flat(config, return_intermediate_keys=self.overwrite_nested_config):
             if self.infer_types and value is not None:
-                parser.add_argument(f'--{key}', type=type(value), default=value)
+                # use type inferencing callable if available, otherwise strictly infer type
+                if type(value) == int or type(value) == float:
```

Comment:
Do you think we should tie the parser to ArgParse Specifically or should we make it generic? 

e.g. 

```python
class TypeInferencer:
    def __init__(self, original_type, strict=False):
        ...
    
    def infer(self, var):
        if self.strict:
            return original_type(var)
        else:
            # infer type smartly
```

Reply:
Where should the inference go if not in between the parser and the config?

Reply:
Right that's what I mean -- we need to have an option to use our own type inference rather than relying on argparse which can be too rigid. It can be just in a separate file (e.g. type_inference.py and just used in the argparse config)